### PR TITLE
chore: Delete helper methods from `UIView+Constraints.swift` - WPB-14398

### PR DESCRIPTION
**Pull request by agisilaos**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-14398" title="WPB-14398" target="_blank">TaskWPB-14398</a>  [iOS] Delete helper methods from `UIView+Constraints.swift`
  </td></table>
  <br />
 

With this PR we get rid of a lot of helper methods from `UIView+Constraints.swift`. We should avoid using those helper methods and instead set the constraints directly. More specifically:

- Removed `func fitIn(view: UIView, inset: CGFloat)` and set constraints directly
- Removed  `Replace fitInConstraints(view: UIView, inset:` and set constraints directly
- Merged  `fitInConstraints` logic into fitIn method to simplify things
- Moved Anchor enum to the file where it's being used

I'm planning to delete `func fitIn(view: UIView, insets: UIEdgeInsets = .zero)` on a separate PR because it's being used in 40 places and it will be harder to review. 

```swift
func fitIn(view: UIView, insets: UIEdgeInsets = .zero) {
        translatesAutoresizingMaskIntoConstraints = false
        NSLayoutConstraint.activate([
            leadingAnchor.constraint(equalTo: view.leadingAnchor, constant: insets.left),
            trailingAnchor.constraint(equalTo: view.trailingAnchor, constant: -insets.right),
            topAnchor.constraint(equalTo: view.topAnchor, constant: insets.top),
            bottomAnchor.constraint(equalTo: view.bottomAnchor, constant: -insets.bottom)
        ])
    }
```

---

### Checklist

- [x] Title contains a reference JIRA issue number like `[WPB-XXX]`.
- [x] Description is filled and free of optional paragraphs.
- [x] Adds/updates automated tests.

---

### CODE REVIEW

File: wire-ios/Wire-iOS/Sources/UserInterface/Components/Views/TextSearchInputView.swift [hunk 1]
```diff
@@ -101,31 +101,34 @@ final class TextSearchInputView: UIView {
...
-                clearButton.heightAnchor.constraint(equalToConstant: StyleKitIcon.Size.tiny.rawValue),
-
-                spinner.trailingAnchor.constraint(equalTo: clearButton.leadingAnchor, constant: -6),
-                spinner.centerYAnchor.constraint(equalTo: clearButton.centerYAnchor),
-                spinner.widthAnchor.constraint(equalToConstant: StyleKitIcon.Size.tiny.rawValue)
-            ]
-        )
+        NSLayoutConstraint.activate([
+            searchInput.leadingAnchor.constraint(equalTo: leadingAnchor, constant: 8),
+            searchInput.trailingAnchor.constraint(equalTo: trailingAnchor, constant: -8),
```
Comment (caldrian) [inline wire-ios/Wire-iOS/Sources/UserInterface/Components/Views/TextSearchInputView.swift:106]: Negative spacing constants always read weird for me.
You also don't set negative values into SwiftUI's `.padding()`, right?

File: wire-ios/Wire-iOS/Sources/UserInterface/Components/Views/TextSearchInputView.swift [hunk 2]
```diff
@@ -101,31 +101,34 @@ final class TextSearchInputView: UIView {
...
-                clearButton.widthAnchor.constraint(equalToConstant: StyleKitIcon.Size.tiny.rawValue),
-                clearButton.heightAnchor.constraint(equalToConstant: StyleKitIcon.Size.tiny.rawValue),
-
-                spinner.trailingAnchor.constraint(equalTo: clearButton.leadingAnchor, constant: -6),
-                spinner.centerYAnchor.constraint(equalTo: clearButton.centerYAnchor),
-                spinner.widthAnchor.constraint(equalToConstant: StyleKitIcon.Size.tiny.rawValue)
-            ]
-        )
+        NSLayoutConstraint.activate([
+            searchInput.leadingAnchor.constraint(equalTo: leadingAnchor, constant: 8),
```
Comment (caldrian) [inline wire-ios/Wire-iOS/Sources/UserInterface/Components/Views/TextSearchInputView.swift:105]: We should consider using the methods which use the system spacing, like https://developer.apple.com/documentation/uikit/nslayoutxaxisanchor/2866112-constraint.

**Review by caldrian — APPROVED** [review]

**Review by jullianm — APPROVED** [review]

File: wire-ios/Wire-iOS/Sources/UserInterface/Components/Views/TextSearchInputView.swift (hunk 2 above)
Comment (agisilaos) [reply wire-ios/Wire-iOS/Sources/UserInterface/Components/Views/TextSearchInputView.swift:105]: We should definitely add it to the Conventions doc. Can you please do so @caldrian?

File: wire-ios/Wire-iOS/Sources/UserInterface/Components/Views/TextSearchInputView.swift (hunk 1 above)
Comment (agisilaos) [reply wire-ios/Wire-iOS/Sources/UserInterface/Components/Views/TextSearchInputView.swift:106]: I don't remember using any negative values on SwiftUI padding. I think it's another great candidate for Conventions. Let's discuss tomorrow!

**Pull request merged**